lib/longleaf/helpers/selection_options_parser.rb

```diff
@@ -0,0 +1,162 @@
+require 'longleaf/candidates/file_selector'
+require 'longleaf/candidates/registered_file_selector'
+require 'longleaf/candidates/manifest_digest_provider'
+require 'longleaf/candidates/single_digest_provider'
+
+module Longleaf
+  # Helper for parsing manifest inputs used for registration
+  class SelectionOptionsParser
+    extend Longleaf::Logging
+
+    # Parses the provided options to construct a file selector and digest provider for
+    # use in registration commands.
+    # @param options [Hash] command options
+    # @param app_config_manager [ApplicationConfigManager] app config manager
+    # @return The file select and digest provider.
+    def self.parse_registration_selection_options(options, app_config_manager)
+      there_can_only_be_one("Only one of the following selection options may be provided: -m, -f, -s",
+          options, :file, :manifest, :location)
+
+      if !options[:manifest].nil?
+        path_to_digests = self.manifests_to_digest_mapping(options[:manifest])
+        selector = FileSelector.new(file_paths: path_to_digests.keys, app_config: app_config_manager)
+        digest_provider = ManifestDigestProvider.new(path_to_digests)
+      elsif !options[:file].nil?
+        if options[:checksums]
+          checksums = options[:checksums]
+          # validate checksum list format, must a comma delimited list of prefix:checksums
+          if /^[^:,]+:[^:,]+(,[^:,]+:[^:,]+)*$/.match(checksums)
+            # convert checksum list into hash with prefix as key
+            checksums = Hash[*checksums.split(/\s*[:,]\s*/)]
+            digest_provider = SingleDigestProvider.new(checksums)
+          else
+            logger.failure("Invalid checksums parameter format, see `longleaf help <command>` for more information")
+            exit 1
+          end
+        end
+
+        file_paths = options[:file].split(/\s*,\s*/)
+        selector = FileSelector.new(file_paths: file_paths, app_config: app_config_manager)
+      elsif !options[:location].nil?
+        storage_locations = options[:location].split(/\s*,\s*/)
+        selector = FileSelector.new(storage_locations: storage_locations, app_config: app_config_manager)
+        digest_provider = SingleDigestProvider.new(nil)
+      else
+        logger.failure("Must provide one of the following file selection options: -f, l, or -m")
+        exit 1
+      end
+
+      [selector, digest_provider]
+    end
+
+    def self.there_can_only_be_one(failure_msg, options, *names)
```

Comment:
Are you sure you don't want to name this `there_can_be_only_one`? https://highlander.fandom.com/wiki/There_can_be_only_one

Reply:
oh man, good catch, I was attempting referencing it but didn't look up the quote. Very embarrassing.